goas/atomic_asset_transfer/risc0_proofs/zone_merge/src/main.rs

```diff
@@ -0,0 +1,44 @@
+use cl::{merkle, nullifier::Nullifier, PtxRoot};
+use goas_proof_statements::zone_funds::MergePrivate;
+use ledger_proof_statements::death_constraint::DeathConstraintPublic;
+use risc0_zkvm::guest::env;
+
+fn main() {
+    let MergePrivate {
+        funds_note,
+        zone_note,
+        merge_event,
+        merge_event_state_path,
+        txs_root,
+        balances_root,
+    } = env::read();
+
+    let input_root = funds_note.input_root();
+    let output_root = zone_note.output_root();
+    let nf = Nullifier::new(funds_note.input.nf_sk, funds_note.input.nonce);
+    // check the zone funds note is the one in the spend event
+    assert_eq!(nf, merge_event.nf);
+
+    // ** Assert merge spent event was an output of the correct zone stf **
+    // The zone state field is a merkle tree over:
+    //                  root
+    //              /        \
+    //            io          state
+    //          /   \        /     \
+    //      events   txs   zoneid  balances
```

Comment:
nit: We have this comment duplicated across a few places, I fear it will get outdated quickly when the zone state schema changes.

Reply:
What do you think of a `common::state_root(event_root, txs_root, zone_id, balances_root)` helper that will define the state schema and we can centralize this comment there.